cmd/demo/main.go

```diff
@@ -11,132 +12,156 @@ import (
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure/auth"
-	"github.com/spf13/pflag"
 	"k8s.io/klog/v2"
 )
 
+const (
+	contextTimeout = 80 * time.Second
+)
+
 var (
-	retryWaitTime  = pflag.Int("retry-wait-time", 20, "retry wait time in seconds")
-	resource       = pflag.String("aad-resourcename", "https://management.azure.com/", "name of resource to grant token")
-	subscriptionID = pflag.String("subscriptionid", "", "subscription id for test")
-	clientID       = pflag.String("clientid", "", "client id for the msi id")
-	resourceGroup  = pflag.String("resourcegroup", "", "any resource group name with reader permission to the aad object")
+	period           time.Duration
+	resourceName     string
+	subscriptionID   string
+	resourceGroup    string
+	identityClientID string
 )
 
 func main() {
-	pflag.Parse()
+	flag.DurationVar(&period, "period", 100*time.Second, "The period that the demo is being executed")
+	flag.StringVar(&resourceName, "resource-name", "https://management.azure.com/", "The resource name to grant the access token")
+	flag.StringVar(&subscriptionID, "subscription-id", "", "The Azure subscription ID")
+	flag.StringVar(&resourceGroup, "resource-group", "", "The resource group name which the user-assigned identity read access to")
+	flag.StringVar(&identityClientID, "identity-client-id", "", "The user-assigned identity client ID")
+	flag.Parse()
 
 	podname := os.Getenv("MY_POD_NAME")
 	podnamespace := os.Getenv("MY_POD_NAMESPACE")
 	podip := os.Getenv("MY_POD_IP")
 
-	klog.Infof("starting demo pod %s/%s %s", podnamespace, podname, podip)
+	klog.Infof("starting aad-pod-identity demo pod (namespace: %s, name: %s, IP: %s)", podnamespace, podname, podip)
 
-	msiEndpoint, err := adal.GetMSIVMEndpoint()
+	imdsTokenEndpoint, err := adal.GetMSIVMEndpoint()
 	if err != nil {
-		klog.Fatalf("failed to get MSI endpoint, error: %+v", err)
+		klog.Fatalf("failed to get IMDS token endpoint, error: %+v", err)
 	}
 
-	for {
-		doARMOperations(*subscriptionID, *resourceGroup)
-
-		t1 := testMSIEndpoint(msiEndpoint, *resource)
-		if t1 == nil {
-			continue
-		}
-
-		t2 := testMSIEndpointFromUserAssignedID(msiEndpoint, *clientID, *resource)
-		if t2 == nil {
-			continue
-		}
+	ticker := time.NewTicker(period)
+	defer ticker.Stop()
 
-		if !strings.EqualFold(t1.AccessToken, t2.AccessToken) {
-			klog.Errorf("msi, emsi test failed with t1(%+v) and t2(%+v)", t1, t2)
+	for {
+		<-ticker.C
```

Comment:
The ticker will not tick immediately. The first tick is only after the `period` is elapsed which by default will be 100s  here. You would need to run the logic first and then wait on the ticker.